src/mlpack/methods/ann/layer/adaptive_max_pooling.hpp

```diff
@@ -0,0 +1,172 @@
+  void Backward(const arma::Mat<eT>& input,
+                arma::Mat<eT>& gy,
+                arma::Mat<eT>& g);
+
+  //! Get the output parameter.
+  const OutputDataType& OutputParameter() const
+  { return poolingLayer.OutputParameter(); }
+
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return poolingLayer.OutputParameter(); }
+
+  //! Get the delta.
+  const OutputDataType& Delta() const { return poolingLayer.Delta(); }
+  //! Modify the delta.
+  OutputDataType& Delta() { return poolingLayer.Delta(); }
+
+  //! Get the width.
+  size_t InputWidth() const { return poolingLayer.InputWidth(); }
+  //! Modify the width.
+  size_t& InputWidth() { return poolingLayer.InputWidth(); }
+
+  //! Get the height.
+  size_t InputHeight() const { return poolingLayer.InputHeight(); }
+  //! Modify the height.
+  size_t& InputHeight() { return poolingLayer.InputHeight(); }
+
+  //! Get the width.
+  size_t OutputWidth() const { return outputWidth; }
+  //! Modify the width.
+  size_t& OutputWidth() { return outputHeight; }
+
+  //! Get the height.
+  size_t OutputHeight() const { return poolingLayer.OutputHeight(); }
+  //! Modify the height.
+  size_t& OutputHeight() { return poolingLayer.OutputHeight(); }
+
+  //! Get the reset parameter of the layer.
+  bool Reset() const { return reset; }
+  //! Modify the reset parameter of the layer.
+  bool& Reset() { return reset; }
+
+  //! Get the input size.
+  size_t InputSize() const { return poolingLayer.InputSize(); }
+
+  //! Get the output size.
+  size_t OutputSize() const { return poolingLayer.OutputSize(); }
+
+  /**
+   * Serialize the layer
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int version);
+
+ private:
+  /**
+   * Initialize Kernel Size and Stride for Adaptive Pooling.
+   */
+  void IntializeAdaptivePadding()
+  {
+    poolingLayer.StrideWidth() = std::floor(poolingLayer.InputWidth() /
+        outputWidth);
+    poolingLayer.StrideHeight() = std::floor(poolingLayer.InputHeight() /
+        outputHeight);
+
+    poolingLayer.KernelWidth() = poolingLayer.InputWidth() -
+        (outputWidth - 1) * poolingLayer.StrideWidth();
+    poolingLayer.KernelHeight() = poolingLayer.InputHeight() -
+        (outputHeight - 1) * poolingLayer.StrideHeight();
+
+    if (poolingLayer.KernelHeight() <= 0 || poolingLayer.KernelWidth() <= 0 ||
+        poolingLayer.StrideWidth() <=0 || poolingLayer.StrideHeight() <= 0)
+    {
+      Log::Fatal << "Given output shape (" << outputWidth << ", "
+        << outputHeight << ") is not possible for given input shape ("
+        << poolingLayer.InputWidth() <<", "<< poolingLayer.InputHeight()
+        << ")."<< std::endl;
+    }
+  }
+
+  //! Locally stored MaxPooling Object.
+  ann::MaxPooling<> poolingLayer;
+
+  //! Locally-stored output width.
+  size_t outputWidth;
+
+  //! Locally-stored output height.
+  size_t outputHeight;
+
+  //! Locally-stored boolean to reset the layer.
```

Comment:
I think this should be //! Locally-stored reset parameter used to initialize the layer once.

Reply:
Thanks for the suggestion. That is much better.